cache/sampleStore.js

```diff
@@ -112,26 +112,28 @@ function removeNullsAndStringifyArrays(obj, arrayFields) {
  * ie. input value: Mon Apr 03 2017 14:10:57 GMT-0700 (PDT)
  * output value: 2017-03-14T02:22:42.255Z
  *
- * @param {Object} Contains keys whose values need be converted
+ * @param {Object} obj - Contains keys whose values need be converted
  * If value is provided, return the ISO formatted date.
  * If no value, return the ISO formatted date with now time.
+ * @returns {String} The date string in ISO format.
+
  */
 function convertToISO(obj) {
-  let key = '';
-  for (let j = constants.ISOfields.length - 1; j >= 0; j--) {
-    key = constants.ISOfields[j];
-    obj[key] = obj[key] ? obj[key].toISOString() :
-      new Date().toISOString();
-  }
-
+  constants.ISOfields.forEach((field) => {
+    if (!obj[field]) {
+      obj[field] = new Date().toISOString();
+    } else if (obj[field] && obj[field].toISOString) {
```

Comment:
Nevermind, I just checked obj[field].toISOString will return undefined for something other than Date.